### PR TITLE
Update WritingStyleGuide.rst for Extensions style

### PR DESCRIPTION
**Pull request by flicstar**
See conversation thread in t3a-content-group Slack channel: https://typo3.slack.com/archives/CJ87LCKT9/p1731995664402719

### CODE REVIEW

**Comment by flicstar** [comment]
@mabolek Please review and make changes. I added in a point about writing extension names in highly technical and procedural docs - do you think that is a context where using the key is acceptable?

File: Documentation/Content/WritingStyleGuide.rst [hunk 1]
```diff
@@ -108,6 +108,33 @@ Examples:
...
+All extensions have a proper name in addition to the lowercase extension key. For example:
+
+- Name: AWS SDK for PHP
+- Key: aws_sdk_php
+
+When we write about extensions, we use the proper name with the convention: "The [proper name] extension...". 
+
+On the first instance of the extension name in the text, we include the key in paranthesis immediately following the proper name. For example, "The AWS SDK for PHP (aws_sdk_php) extension...". You only need to do this once. 
+
+Where appropriate, on the first instance of the extension name in the text, we also include the link to the extension's page in the TYPO3 Extension Repository `(extensions.typo3.org) <https://extensions.typo3.org/>`__.
```
Comment (mabolek) [inline Documentation/Content/WritingStyleGuide.rst]:
```suggestion
Where appropriate, on the first instance of the extension name in the text, we also include the link to the extension's page in the `TYPO3 Extension Repository <https://extensions.typo3.org/>`__.
```

File: Documentation/Content/WritingStyleGuide.rst [hunk 2]
```diff
@@ -108,6 +108,33 @@ Examples:
...
+==========
+
+All extensions have a proper name in addition to the lowercase extension key. For example:
+
+- Name: AWS SDK for PHP
+- Key: aws_sdk_php
+
+When we write about extensions, we use the proper name with the convention: "The [proper name] extension...". 
+
+On the first instance of the extension name in the text, we include the key in paranthesis immediately following the proper name. For example, "The AWS SDK for PHP (aws_sdk_php) extension...". You only need to do this once.
```
Comment (mabolek) [inline Documentation/Content/WritingStyleGuide.rst]:
```suggestion
On the first instance of the extension name in the text, we include the key in paranthesis immediately following the proper name. For example, "The AWS SDK for PHP (aws_sdk_php) extension...".
```

File: Documentation/Content/WritingStyleGuide.rst [hunk 3]
```diff
@@ -108,6 +108,33 @@ Examples:
...
 
+.. _extensions:
+
+Extensions
+==========
+
+All extensions have a proper name in addition to the lowercase extension key. For example:
+
+- Name: AWS SDK for PHP
+- Key: aws_sdk_php
```
Comment (mabolek) [inline Documentation/Content/WritingStyleGuide.rst]:
```suggestion
* Name: AWS SDK for PHP
* Key: aws_sdk_php
```

**Review by mabolek — CHANGES_REQUESTED** [review]
Looks nice. I made some suggested adjustments.

**Review by mabolek — APPROVED** [review]

**Pull request merged**